spec/features/user_enters_valid_email_spec.rb

```diff
@@ -1,5 +1,39 @@
 require 'rails_helper'
 
-RSpec.feature 'Sign-up email', type: :feature do
-  
+RSpec.feature 'Sign-in email', type: :feature do
+  scenario "user enter blank password with valid e-mail" do
+    sign_up_with_sign_out
+    fill_in('Email', :with => 'user@gmail.com')
+    click_button "Log in"
+    expect(page).to have_content("Invalid Email or password.")
+  end
+
+  scenario "user enter blank password and blank e-mail" do
+    sign_up_with_sign_out
+    click_button "Log in"
+    expect(page).to have_content("Invalid Email or password.")
+  end
+
+  scenario "user enter blank e-mail with valid password" do
+    sign_up_with_sign_out
+    fill_in('Password', :with => '123456')
+    click_button "Log in"
+    expect(page).to have_content("Invalid Email or password.")
+  end
+
+  scenario "user enter valid e-mail with incorrect password" do
+    sign_up_with_sign_out
+    fill_in('Email', :with => 'user@gmail.com')
```

Comment:
Tests look great but for these last tow tests it isn't immediately obvious that the user is putting in anything wrong because we can't see them log in, two proposed solutions:

- put a comment at the beginning of these tests stating the correct email and password
- have the sign_up_with_sign_out method take the email and password as parameters so it is clear what the correct details are in the test

Reply:
I agree with this - in the interests of making the tests super readable